contracts/EVMScriptFactories/CSMSettleELStealingPenalty.sol

```diff
@@ -0,0 +1,109 @@
+
+    // -------------
+    // ERRORS
+    // -------------
+
+    string private constant ERROR_EMPTY_NODE_OPERATORS_IDS =
+        "EMPTY_NODE_OPERATORS_IDS";
+    string private constant ERROR_OUT_OF_RANGE_NODE_OPERATOR_ID =
+        "OUT_OF_RANGE_NODE_OPERATOR_ID";
+    string private constant ERROR_NOT_SORTED_NODE_OPERATORS_IDS =
+        "NOT_SORTED_NODE_OPERATORS_IDS";
+
+    // -------------
+    // VARIABLES
+    // -------------
+
+    /// @notice Address of CSModule
+    ICSModule public immutable csm;
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(address _trustedCaller, address _csm)
+        TrustedCaller(_trustedCaller)
+    {
+        csm = ICSModule(_csm);
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Creates EVMScript to settle EL stealing penalty for a specific node operators on CSM
+    /// @param _creator Address who creates EVMScript
+    /// @param _evmScriptCallData Encoded: uint256[] memory nodeOperatorIds
+    function createEVMScript(address _creator, bytes memory _evmScriptCallData)
+        external
+        view
+        override
+        onlyTrustedCaller(_creator)
+        returns (bytes memory)
+    {
+        uint256[] memory nodeOperatorIds = _decodeEVMScriptCallData(_evmScriptCallData);
+
+        _validateInputData(nodeOperatorIds);
+
+        return
+            EVMScriptCreator.createEVMScript(
+                address(csm),
+                ICSModule.settleELRewardsStealingPenalty.selector,
+                _evmScriptCallData
+            );
+    }
+
+    /// @notice Decodes call data used by createEVMScript method
+    /// @param _evmScriptCallData Encoded: uint256[] memory nodeOperatorIds
+    /// @return Node operator IDs to settle EL stealing penalty
+    function decodeEVMScriptCallData(bytes memory _evmScriptCallData)
+        external
+        pure
+        returns (uint256[] memory)
+    {
+        return _decodeEVMScriptCallData(_evmScriptCallData);
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    function _decodeEVMScriptCallData(bytes memory _evmScriptCallData)
+        private
+        pure
+        returns (uint256[] memory)
+    {
+        return abi.decode(_evmScriptCallData, (uint256[]));
+    }
+
+    function _validateInputData(
+        uint256[] memory _decodedCallData
+    ) private view {
+        uint256 nodeOperatorsCount = csm.getNodeOperatorsCount();
+        require(_decodedCallData.length > 0, ERROR_EMPTY_NODE_OPERATORS_IDS);
+        require(
+            _decodedCallData[_decodedCallData.length - 1] < nodeOperatorsCount,
+            ERROR_OUT_OF_RANGE_NODE_OPERATOR_ID
+        );
+        for (uint256 i = 0; i < _decodedCallData.length; ++i) {
+            require(
+                i == 0 ||
+                    _decodedCallData[i] > _decodedCallData[i - 1],
+                ERROR_NOT_SORTED_NODE_OPERATORS_IDS
```

Comment:
Let's ignore sorting requirement (because there's no such requirement in fact), and just check all the values against `nodeOperatorsCount`.

Reply:
I guess it is done to check max noId

Reply:
fixed